packages/api/src/command/settings/updateSettings.ts

```diff
@@ -3,28 +3,37 @@ import { nanoid } from "nanoid";
 import { maxWebhookStatusLength } from "../../domain/settings";
 import { processAsyncError } from "../../errors";
 import WebhookError from "../../errors/webhook";
-import { Settings, WEBHOOK_STATUS_BAD_RESPONSE, WEBHOOK_STATUS_OK } from "../../models/settings";
+import {
+  Settings as SettingsModel,
+  WEBHOOK_STATUS_BAD_RESPONSE,
+  WEBHOOK_STATUS_OK,
+} from "../../models/settings";
 import { capture } from "../../shared/notifications";
 import { Util } from "../../shared/util";
 import { errorToWhStatusDetails, sendTestPayload } from "../webhook/webhook";
 import { getSettingsOrFail } from "./getSettings";
+import { MrFilters } from "../../domain/settings";
 
 const log = Util.log(`updateSettings`);
 
 export type UpdateSettingsCommand = {
   cxId: string;
   webhookUrl?: string;
+  filters?: MrFilters[];
 };
 
 export const updateSettings = async ({
   cxId,
   webhookUrl,
-}: UpdateSettingsCommand): Promise<Settings> => {
+  filters,
+}: UpdateSettingsCommand): Promise<SettingsModel> => {
   const originalSettings = await getSettingsOrFail({ id: cxId });
   const updateWebhook = getWebhookDataForUpdate(originalSettings, webhookUrl);
-  await Settings.update(
+  await SettingsModel.update(
     {
-      ...updateWebhook,
+      ...originalSettings.dataValues,
```

Comment:
This doesn't update only fields that are mentioned? I'd test, but do we need `...originalSettings.dataValues,`?

Reply:
Yea i did test it by removing the webhookUrl and making sure it didnt ovveride it

Reply:
Ideally it's be something like:
```
originalSettings.update({
    webhookUrl,
    mrFilters,
})
```

Where if they're undefined, the current value doesn't change. I believe to wipe an existing value for a nullable field you'd need to pass in *null* -- undefined should not perform an update.

Reply:
Can think about it in the refactor.